File: LC03.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) in [0,1]:
            return len(s)
        head, tail, answer = 0, 1, 0
        hashset = set()
        hashset.add(s[head])
        while tail < len(s):
            if s[tail] in hashset:
                answer = len(hashset) if len(hashset) > answer else answer
                hashset.remove(s[head])
                head += 1
            else:
                hashset.add(s[tail])
                tail += 1
        return answer if answer > len(hashset) else len(hashset)

File: LC03.java
import java.util.*;

public class question_3 {
    class Solution {
        public int lengthOfLongestSubstring(String s) {
            int head = 0,tail = 1;
            HashSet<Character> set = new HashSet<Character>();
            int answer = 0;
            if(s.isEmpty()) return answer;
            set.add(s.charAt(head));
            while(tail < s.length()){
                if (!set.add(s.charAt(tail))){
                    answer = set.size() > answer? set.size() : answer;
                    set.remove(s.charAt(head));
                    head++;
                }else{
                    tail++;
                }
            }
            return answer;
        }
    }
}
